### ISSUE TITLE
Error [ERR_UNHANDLED_ERROR]: Unhandled error. ([object Object])

### ISSUE THREAD

**Issue by rafael-trevisan** [OraOpenSource/apex-nitro#245]
Hi,

I was trying APEX Nitro version 4.1.0-beta.4 and it's raising a constant error whenever I save any `.scss`

Saving `.js` is fine, the problem only happens when saving `.scss`

That may help:

```
$ sass --version
Sass 3.5.5 (Bleeding Edge)
```

Here's the message on the terminal when the problem occurs.
 
```
/usr/local/lib/node_modules/apex-nitro/node_modules/async-done/index.js:18
    throw err;
    ^

Error [ERR_UNHANDLED_ERROR]: Unhandled error. ([object Object])
    at Gulp.emit (events.js:141:19)
    at Object.error (/usr/local/lib/node_modules/apex-nitro/node_modules/undertaker/lib/helpers/createExtensions.js:61:10)
    at handler (/usr/local/lib/node_modules/apex-nitro/node_modules/now-and-later/lib/map.js:46:14)
    at f (/usr/local/lib/node_modules/apex-nitro/node_modules/once/once.js:25:25)
    at f (/usr/local/lib/node_modules/apex-nitro/node_modules/once/once.js:25:25)
    at tryCatch (/usr/local/lib/node_modules/apex-nitro/node_modules/async-done/index.js:24:15)
    at done (/usr/local/lib/node_modules/apex-nitro/node_modules/async-done/index.js:40:12)
    at Domain.onError (/usr/local/lib/node_modules/apex-nitro/node_modules/async-done/index.js:51:5)
    at Object.onceWrapper (events.js:255:19)
    at Domain.emit (events.js:160:13)

```

And here's is json for this project:

```
{
  "mode": "advanced",
  "appURL": "https://apex.oracle.com/pls/apex/f?p=***",
  "srcFolder": "***/***/src",
  "distFolder": "***/***/dist",
  "js": {
    "processor": "webpack",
    "concat": false,
    "entries": ["***/***/src/js/***.js"],
    "bundleFilename": "***.bundle",
    "library": true,
    "libraryName": "***",
    "tsConcat": false
  },
  "css": {
    "language": "sass",
    "concat": false
  },
  "browsersync": {
    "notify": false,
    "ghostMode": false
  },
  "header": {
    "enabled": false
  },
  "apex": {
    "openBuilder": true
  },
  "publish": {
    "destination": "application"
  }
}
```

Cheers

**Comment by rafael-trevisan** [comment]
Forgot to mention: switching back to APEX Nitro version 4.0.0 solves the issue

**Comment by vincentmorneau** [comment]
Thanks for reporting this. If I'm not mistaken, APEX Nitro still compiles your Sass files, but then crashes when watching for changes, right?

It has been fixed in our development environment already. We are planning on releasing the official 4.1.0 early this week.

**Comment by vincentmorneau** [comment]
It's published. Can you update and try again? 

We're going to make an official announcement on Thursday, so I'm trying to avoid social media posts until then :)

**Comment by rafael-trevisan** [comment]
Cool! Yes, I can! I'll do that later and will let you know.

Got it 😉 

Thanks

**Comment by rafael-trevisan** [comment]
Hey @vincentmorneau,

I've upgraded it and haven't got the error anymore 😃 

Everything is working fine!

Just a thing that I meant to comment before but I keep forgetting: On macOS High Sierra (10.13.3) I only got apex-nitro installed if I use `--unsafe-perm` for i.e. 
```
$ sudo npm install -g --unsafe-perm apex-nitro
```
I thought I had to use `--unsafe-perm` due to some node-sass issue and I didn't want to bother you. But now, even using APEX Nitro **version 4.1.0** I still get an error without `--unsafe-perm`.

If you want to dig into this issue let me know. I'll be happy to open a new issue with proper log and steps to reproduce the problem.

Basically, the error starts with
```
gyp verb `which` failed Error: not found: python2
...
...
gyp ERR! stack Error: EACCES: permission denied, mkdir '/usr/local/lib/node_modules/apex-nitro/node_modules/node-sass/build'
...
...
```

I mean, it isn't a big issue as long there's a workaround, but maybe you might want to include that in the troubleshooting notes.

Cheers and thanks for the upgrade! 👍

**Comment by vincentmorneau** [comment]
Hi Rafael,

You bring a good point. I think using `--unsafe-perm` works, but most people would not feel comfortable using this. I found the following: https://docs.npmjs.com/getting-started/fixing-npm-permissions 

Which is the "right" way to fix the EACCES problem. I use `nvm` myself (node version manager), so it's probably why I never encountered the problem. And the node-sass Github page is full of closed issues regarding this problem.

I'll add a troubleshooting note. Thanks!
